### ISSUE TITLE
import/no-unresolved errors if package omits `.js` extension in `main`

### ISSUE THREAD

**Issue by jamesreggio** [benmosher/eslint-plugin-import#617]
If a dependency (for instance, a package in `node_modules`) omits an extension in the `main` field of its `package.json`, the `import/no-unresolved` rule will fail.

For example:

```javascript
// node_modules/foo/package.json
{
  name: "foo",
  version: "1.0.0",
  main: "index"
}

// node_modules/foo/index.js
const message = 'It is legal to omit a .js on "main" in package.json';
export default message;

// index.js
import foo from 'foo'; // This triggers the import/no-unresolved rule
console.log(foo);
```

I imagine this should be a simple fix.

**Comment by benmosher** [comment]
Ah, wild. Thanks for the heads up, didn't realize that was possible. I think you're right, should be a quick fix. 😄

**Comment by elanpang** [comment]
hope it will be fixed
